### PR TITLE
TW-1791: Improve display contacts on multiple homeserver

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket:

- #1791 

### Resolved

- [x] Only display `ToM` contacts for homeserver is supported Twake
- [x] When we switch or add another account need to refresh all contacts (`ToM`, phone book, recent contacts)
- [x] For web only show `ToM` contacts is server supported and recent contacts

#### Mobile

https://github.com/linagora/twake-on-matrix/assets/99852347/c8b57a6e-8b56-4d77-ab5c-ce13ca7b6b05

#### Web

https://github.com/linagora/twake-on-matrix/assets/99852347/4f1e3451-9a53-4765-91d9-9dfdc3b289ec

https://github.com/linagora/twake-on-matrix/assets/99852347/38e95d45-262f-4d25-bdd0-6654cff9f6c1

### CODE REVIEW

File: lib/presentation/mixins/contacts_view_controller_mixin.dart
Comment (Te-Z) [inline lib/presentation/mixins/contacts_view_controller_mixin.dart:319]: can you explain why remove those lines ?

File: test/domain/contacts/contacts_manager_test.dart
Comment (Te-Z) [inline test/domain/contacts/contacts_manager_test.dart:1]: Good job with this

**Review by Te-Z — APPROVED** [review]

File: lib/presentation/mixins/contacts_view_controller_mixin.dart
Comment (nqhhdev) [reply lib/presentation/mixins/contacts_view_controller_mixin.dart:319]: I reverted it and write differently that is clearer and easier to understand.

**Comment by sherlockvn** [comment]
i tested on this ticket. It works well
- test case log out
- test multiple account login then log out
- test again direct chat, new group chat
- test contact when switch between server

**Review by sherlockvn — APPROVED** [review]

**Review by hieutbui — APPROVED** [review]

**Comment by Te-Z** [comment]
LGTM, please resolve conflict

**Comment by nqhhdev** [comment]
> LGTM, please resolve conflict

DONE

**Comment by Te-Z** [comment]
@hoangdat I can't merge this one. It says it requires 2 approvals and there's already 3

**Review by Te-Z — APPROVED** [review]

**Pull request merged**
